tensorflow/contrib/distribute/python/checkpoint_utils_test.py

```diff
@@ -91,6 +95,36 @@ def init_and_verify(g):
       else:
         init_and_verify(g)
 
+  @combinations.generate(combinations.combine(
+      distribution=[combinations.default_strategy,
+                    combinations.one_device_strategy,
+                    combinations.mirrored_strategy_with_gpu_and_cpu,
+                    combinations.mirrored_strategy_with_two_gpus,
+                    combinations.core_mirrored_strategy_with_gpu_and_cpu,
+                    combinations.core_mirrored_strategy_with_two_gpus],
+      in_replica_mode=[True, False],
+      mode=["graph"]))
+  def testInitFromDifferentNameObject(self, distribution, in_replica_mode):
+    checkpoint_dir, v1_value, _ = self._get_test_object()
+
+    def init_and_verify(g):
+      v1 = variable_scope.get_variable("new_var1", [1, 10])
+      # Use string add to create new object in each replica
+      prefix = "new_"
+      suffix = "var1"
+      new_var1 = prefix + suffix
+      checkpoint_utils.init_from_checkpoint(checkpoint_dir, {
+          "var1": new_var1,
+      })
+      with self.test_session(graph=g) as session:
+        session.run(variables.global_variables_initializer())
+        self.assertAllEqual(v1_value, self.evaluate(v1))
+
+    with ops.Graph().as_default() as g, distribution.scope():
+      if in_replica_mode:
+        distribution.extended.call_for_each_replica(
+            init_and_verify, [g])
+
```

Comment:
seems like missing the `else` case?

Reply:
Non-distributed case is unnecessary to test so I don't add else here.

Reply:
Oh i see. if you don't want to test that case, you should remove the `in_replica_mode` from the combinations and arguments for this test?

Reply:
I have already added the non-distributed test case in the latest commits.